### PR TITLE
Marcos.reginato/styles and themes

### PR DESCRIPTION
**Pull request by regmarcos**
Hi Guys! In this PR i added the landing screen and improves the layout using styles and using a different font than the default font from Android. Also, i added a toolbar with the MARVEL logo.

Here is the evidence from that changes:
https://gfycat.com/realisticrelievedafricanwildcat

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -48,6 +48,8 @@ dependencies {
             appDependencies.cardView,
             appDependencies.material
 
+    implementation 'androidx.appcompat:appcompat:1.0.2'
```
Comment (SartiMau) [inline app/build.gradle]: remove this and add the dependencies to the generalDependencies, just like the others pls

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -14,7 +13,15 @@
...
+        <activity android:name=".presentation.MarvelLandingScreen"
+            android:theme="@style/LandingScreen">
+            <intent-filter>
+            <action android:name="android.intent.action.MAIN" />
+            <category android:name="android.intent.category.LAUNCHER" />
+            </intent-filter>
+        </activity>
+        <activity android:name=".presentation.MainActivity"
+            android:theme="@style/Theme.AppCompat.NoActionBar">
             <intent-filter>
```
Comment (SartiMau) [inline app/src/main/AndroidManifest.xml]: remove this <intent-filter>. we only have one LAUNCHER screen and, in your case, is not MainActivity anymore, now it is LandingScreen

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/MarvelLandingScreen.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.os.Handler
+import android.view.animation.AlphaAnimation
+import com.puzzlebench.clean_marvel_kotlin.R
+import kotlinx.android.synthetic.main.activity_marvel_landing_screen.logo_marvel
+import kotlinx.android.synthetic.main.activity_marvel_landing_screen.progress_bar_landing_screen
+
+class MarvelLandingScreen : AppCompatActivity() {
```
Comment (SartiMau) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/MarvelLandingScreen.kt]: rename the activity as MarvelLandingScreenActivity pls

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/MarvelLandingScreen.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    private val duration = 2000
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_marvel_landing_screen)
+        setAnimations()
+    }
+
+    private fun setAnimations() {
```
Comment (SartiMau) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/MarvelLandingScreen.kt]: this method must be in the View.. so, from the activity you will call the presenter and from the presenter, you will call the view that is who will have this method

File: app/src/main/res/layout-land/activity_main.xml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+    <androidx.appcompat.widget.Toolbar
+        android:id="@+id/toolbar"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="?attr/colorPrimary"
+        android:minHeight="?attr/actionBarSize"
+        android:theme="@style/ThemeOverlay.AppCompat.ActionBar"
+        app:layout_constraintTop_toTopOf="parent"
+        app:title="Marvel APP">
```
Comment (SartiMau) [inline app/src/main/res/layout-land/activity_main.xml]: no hardcode pls

File: app/src/main/res/layout-land/activity_main.xml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+        <ImageView
+            android:layout_width="100dp"
```
Comment (SartiMau) [inline app/src/main/res/layout-land/activity_main.xml]: no hardcode pls

File: app/src/main/res/layout-land/activity_main.xml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+            android:layout_height="50dp"
```
Comment (SartiMau) [inline app/src/main/res/layout-land/activity_main.xml]: no hardcode pls

File: app/src/main/res/layout-land/activity_main.xml [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        android:layout_height="0dp"
+        app:layout_constraintTop_toBottomOf="@id/toolbar"
+        tools:layout_editor_absoluteX="0dp" />
+
+    <com.google.android.material.floatingactionbutton.FloatingActionButton
+        android:id="@+id/refresh_fab"
+        style="@style/Marvel.FloatingActionButton"
+        android:layout_gravity="bottom|end"
+        android:src="@android:drawable/stat_notify_sync"
+        app:borderWidth="0dp"
```
Comment (SartiMau) [inline app/src/main/res/layout-land/activity_main.xml]: no hardcode pls

File: app/src/main/res/layout-land/activity_main.xml [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toStartOf="@id/database_fab"
+        app:layout_constraintStart_toStartOf="parent" />
+
+    <com.google.android.material.floatingactionbutton.FloatingActionButton
+        android:id="@+id/database_fab"
+        style="@style/Marvel.FloatingActionButton"
+        android:layout_gravity="bottom|start"
+        android:src="@drawable/database_fab_icon"
+        app:borderWidth="0dp"
```
Comment (SartiMau) [inline app/src/main/res/layout-land/activity_main.xml]: no hardcode pls

File: app/src/main/res/layout/activity_main.xml [hunk 10]
```diff
@@ -1,56 +1,71 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     tools:context=".presentation.MainActivity">
 
+    <androidx.appcompat.widget.Toolbar
+        android:id="@+id/toolbar"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="?attr/colorPrimary"
+        android:minHeight="?attr/actionBarSize"
+        app:title="Marvel APP"
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_main.xml]: same

File: app/src/main/res/layout/activity_main.xml [hunk 11]
```diff
@@ -1,56 +1,71 @@
...
+        android:theme="@style/ThemeOverlay.AppCompat.ActionBar"
+        app:layout_constraintTop_toTopOf="parent">
+
+        <ImageView
+           android:layout_width="100dp"
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_main.xml]: same

File: app/src/main/res/layout/activity_main.xml [hunk 12]
```diff
@@ -1,56 +1,71 @@
...
+           android:layout_height="50dp"
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_main.xml]: same

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/MarvelLandingScreen.kt [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.os.Handler
+import android.view.animation.AlphaAnimation
+import com.puzzlebench.clean_marvel_kotlin.R
+import kotlinx.android.synthetic.main.activity_marvel_landing_screen.logo_marvel
+import kotlinx.android.synthetic.main.activity_marvel_landing_screen.progress_bar_landing_screen
+
+class MarvelLandingScreen : AppCompatActivity() {
+    private val duration = 2000
```
Comment (lucianoeli) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/MarvelLandingScreen.kt]: could use a constant, and make it Long so you dont  need to convert it every time. (2000L)

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/MarvelLandingScreen.kt [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_marvel_landing_screen)
+        setAnimations()
+    }
+
+    private fun setAnimations() {
+        progress_bar_landing_screen.isIndeterminate = true
+        val fadeIn = AlphaAnimation(0f, 1f)
```
Comment (SartiMau) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/MarvelLandingScreen.kt]: no hardcode pls

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/MarvelLandingScreen.kt (hunk 13 above)
Comment (SartiMau) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/MarvelLandingScreen.kt]: and put this constant in the view too pls

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 1 above)
Comment (regmarcos) [reply app/build.gradle]: done

File: app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (regmarcos) [reply app/src/main/AndroidManifest.xml]: done

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/MarvelLandingScreen.kt (hunk 3 above)
Comment (regmarcos) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/MarvelLandingScreen.kt]: done

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/MarvelLandingScreen.kt (hunk 13 above)
Comment (regmarcos) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/MarvelLandingScreen.kt]: done

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/MarvelLandingScreen.kt (hunk 4 above)
Comment (regmarcos) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/MarvelLandingScreen.kt]: done

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/MarvelLandingScreen.kt (hunk 14 above)
Comment (regmarcos) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/MarvelLandingScreen.kt]: done

File: app/src/main/res/layout/activity_main.xml (hunk 12 above)
Comment (regmarcos) [reply app/src/main/res/layout/activity_main.xml]: done

File: app/src/main/res/layout/activity_main.xml (hunk 11 above)
Comment (regmarcos) [reply app/src/main/res/layout/activity_main.xml]: done

File: app/src/main/res/layout/activity_main.xml (hunk 10 above)
Comment (regmarcos) [reply app/src/main/res/layout/activity_main.xml]: done

File: app/src/main/res/layout-land/activity_main.xml (hunk 9 above)
Comment (regmarcos) [reply app/src/main/res/layout-land/activity_main.xml]: done

File: app/src/main/res/layout-land/activity_main.xml (hunk 8 above)
Comment (regmarcos) [reply app/src/main/res/layout-land/activity_main.xml]: done

File: app/src/main/res/layout-land/activity_main.xml (hunk 5 above)
Comment (regmarcos) [reply app/src/main/res/layout-land/activity_main.xml]: done

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/view/MarvelLandingScreenView.kt [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+class MarvelLandingScreenView(activity: MarvelLandingScreenActivity) : MarvelLandingScreenContracts.view {
+
+    private val activityRef = WeakReference(activity)
+    override fun init() {
+        setAnimations()
+    }
+
+    override fun setAnimations() {
+        activityRef.get().let {
+            it?.progress_bar_landing_screen?.isIndeterminate = true
```
Comment (SartiMau) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/view/MarvelLandingScreenView.kt]: it is necessary the safe call (it?.) inside the let??

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/presenter/MarvelLandingScreenPresenter.kt [hunk 16]
```diff
@@ -0,0 +1,10 @@
+package com.puzzlebench.clean_marvel_kotlin.presentation.mvp.presenter
+
+import com.puzzlebench.clean_marvel_kotlin.presentation.mvp.MarvelLandingScreenContracts
+import com.puzzlebench.clean_marvel_kotlin.presentation.mvp.view.MarvelLandingScreenView
+
+class MarvelLandingScreenPresenter(val view: MarvelLandingScreenView): MarvelLandingScreenContracts.presenter {
```
Comment (lucianoeli) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/presenter/MarvelLandingScreenPresenter.kt]: if you are using a contract, then you should instantiate the view interface
`...presenter(val view: MarvelLandingScreenContracts.View)` this should avoid problems when you mock the view in unit tests

Comment (regmarcos) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/presenter/MarvelLandingScreenPresenter.kt]: done

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/view/MarvelLandingScreenView.kt (hunk 15 above)
Comment (regmarcos) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/view/MarvelLandingScreenView.kt]: done

**Review by lucianoeli — APPROVED** [review]

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/fragments/mvp/FragmentDialogModel.kt [hunk 17]
```diff
@@ -1,5 +1,6 @@
 package com.puzzlebench.clean_marvel_kotlin.presentation.fragments.mvp
 
+import com.puzzlebench.clean_marvel_kotlin.presentation.fragments.mvp.contracts.FragmentDialogContracts
```
Comment (SartiMau) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/fragments/mvp/FragmentDialogModel.kt:3]: are you using this import? the only change here is this adding

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/fragments/mvp/FragmentDialogView.kt [hunk 18]
```diff
@@ -8,10 +8,10 @@ import android.widget.Toast
 import com.puzzlebench.clean_marvel_kotlin.R
 import com.puzzlebench.clean_marvel_kotlin.presentation.DOT
 import com.puzzlebench.clean_marvel_kotlin.presentation.MainActivity
-import com.puzzlebench.clean_marvel_kotlin.presentation.NO_DESCRIPTION
 import com.puzzlebench.clean_marvel_kotlin.presentation.SCREEN_PERCENTAGE
 import com.puzzlebench.clean_marvel_kotlin.presentation.extension.getImageByUrl
 import com.puzzlebench.clean_marvel_kotlin.presentation.fragments.CharacterFragmentDialog
+import com.puzzlebench.clean_marvel_kotlin.presentation.fragments.mvp.contracts.FragmentDialogContracts
```
Comment (SartiMau) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/fragments/mvp/FragmentDialogView.kt:14]: dame

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/fragments/mvp/FragmentDialogModel.kt (hunk 17 above)
Comment (regmarcos) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/fragments/mvp/FragmentDialogModel.kt:3]: yes, i'm using contracts for the model of FragmentDialogCharacter, is the only model that has a contract.

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/fragments/mvp/FragmentDialogView.kt (hunk 18 above)
Comment (regmarcos) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/fragments/mvp/FragmentDialogView.kt:14]: the view implements the FragmentDialogContracts

**Review by SartiMau — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Pull request merged**